src/metabase/util/malli/fn.clj

```diff
@@ -300,23 +300,12 @@
 ;; ------------------------------ Skipping Namespace Enforcement in prod ------------------------------
 
 (defn instrument-ns?
-  "Used to track namespaces to not enforce malli schemas on with `mu.fn/fn`."
+  "Returns true if mu.fn/fn malli schema validation should happen in production for a given namespace."
   [namespace]
-  (let [lib-and-middleware [#"^metabase\.lib\..*"
-                            #"^metabase\.query-processor\.middleware\..*"
-                            #"^metabase\.sync.*"
-                            #"^metabase\.upload.*"]
-        matches?           (core/fn [namespace regexes]
-                             (let [n (-> namespace ns-name str)]
-                               (some #(re-matches % n) regexes)))
-        ;; empty but placeholder for any namespaces we want to never instrument (in prod)
-        ad-hoc             #{}
-        m                  (meta namespace)]
-    (cond (:instrument/always m)                  true
-          (:instrument/never m)                   false
-          (matches? namespace lib-and-middleware) false
-          (contains? ad-hoc (ns-name namespace))  false
-          :else                                   true)))
+  (let [m (meta namespace)]
+    (cond (:instrument/always m) true
+          (:instrument/never m)  false
+          :else                  true)))
```

Comment:
Doesn't this turn instrumentation _on_ in all cases? Maybe I'm missing some code that does that at a higher level.

Reply:
It indeed looks like the `:else` case is backward here - perhaps got tripped up that `*skip-ns-decision-fn*` just inverts it? Perhaps it's worth adding unit tests for this function.

Reply:
Good eyes, this indeed turns it on. Fixed here daca208c781952ae99c4e7c8020997eab7916a1e I'll write a unit test for `*skip-ns-decision-fn*`.